background.ts

```diff
@@ -388,45 +388,51 @@ Electron.ipcMain.on('k8s-state', (event) => {
   event.returnValue = k8smanager.state;
 });
 
-Electron.ipcMain.on('k8s-reset', async(event, arg) => {
-  try {
-    // If not in a place to restart than skip it
-    if (![K8s.State.STARTED, K8s.State.STOPPED, K8s.State.ERROR].includes(k8smanager.state)) {
-      console.log(`Skipping reset, invalid state ${ k8smanager.state }`);
-
-      return;
-    }
-
-    if (k8smanager.version !== cfg.kubernetes.version ||
-      (await k8smanager.cpus) !== cfg.kubernetes.numberCPUs ||
-      (await k8smanager.memory) !== cfg.kubernetes.memoryInGB * 1024) {
-      arg = 'slow';
-    }
-    switch (arg) {
-    case 'fast':
-      await k8smanager.reset(cfg.kubernetes);
-      break;
-    case 'slow': {
-      await k8smanager.stop();
-
-      console.log(`Stopped Kubernetes backened cleanly.`);
-      console.log('Deleting VM to reset...');
-      await k8smanager.del();
-      console.log(`Deleted VM to reset exited cleanly.`);
+Electron.ipcMain.on('current-port', (event) => {
+  event.returnValue = k8smanager.port;
+});
 
-      // The desired Kubernetes version might have changed
-      k8smanager = newK8sManager();
+Electron.ipcMain.on('k8s-reset', async(_, arg) => {
+  await doK8sReset(arg);
+});
 
-      await k8smanager.start(cfg.kubernetes);
-      break;
-    }
-    default:
-      console.error(`Don't know how to do a ${ arg } reset`);
+async function doK8sReset(optionalArg?: string): Promise<void> {
+  // If not in a place to restart than skip it
+  if (![K8s.State.STARTED, K8s.State.STOPPED, K8s.State.ERROR].includes(k8smanager.state)) {
+    console.log(`Skipping reset, invalid state ${ k8smanager.state }`);
+  } else {
+    try {
+      let arg = (typeof optionalArg === 'undefined' ? '' : optionalArg);
+
+      if (!['slow', 'fast'].includes(arg) &&
+        (k8smanager.version !== cfg.kubernetes.version ||
+          (await k8smanager.cpus) !== cfg.kubernetes.numberCPUs ||
+          (await k8smanager.memory) !== cfg.kubernetes.memoryInGB * 1024 ||
+          (k8smanager.port) !== cfg.kubernetes.port)) {
+        arg = 'slow';
+      }
```

Comment:
This is complicated enough that I wonder if we should expand this a bit… Maybe at least make the `!['slow', 'fast']` part a separate statement?

Reply:
done